### ISSUE TITLE
device.connect gives ringing tone and immediately a disconnect tone, call does not connect.

### ISSUE THREAD

**Issue by ricksonmenezes** [twilio/twilio-voice.js#78]
I have integrated (migrated to)  the 2.1.0 VoiceSDK (JS) version of Twilio but I am unable to get a call to go through. 

I have changed all the old Twilio.Device calls and my code looks like this. (the vm is because we are doing the calls via vuejs)

`var tokenId = "<%=token%>";

                vm.device = new Twilio.Device(tokenId);

                const deviceOptions = {debug: false};
                vm.device.updateOptions(deviceOptions)
                vm.device.register();

                vm.device.on('error', (twilioError, call) => {
                    console.log('An error has occurred: ', + twilioError);
                });

                vm.device.on('registered', device => {
                    console.log('    twilio registered........');
                });

                vm.device.on('registering', device => {
                    console.log('    twilio registering...    ');

            });
                vm.device.on('tokenWillExpire', () => {
               //     const token = getNewTokenViaAjax(); call API to get new token
                    console.log('twilio token expired')
                //device.updateToken(token);
            });
                vm.device.on('unregistered', () => {
                console.log("    twilio unregistered...    ");
            });

                vm.device.on('incoming', conn => {
                console.log("Incoming connection from " + conn.parameters.From);
                conn.accept()
            });

                vm.device.on('connect', connection => {
                    console.log("connection opened " + connection);
                });

                vm.device.on('disconnect', connection => {
                    console.log("connection disonnected " + connection);

                });`

I am able to see the "device registered"  in the console logs to which I assume the token received from the server works fine and twilio registers the device for incoming calls. The twilio version on the server is twilio-java-sdk 3.3.9 jar using the Capability Token API 

```
TwilioCapability capability = new TwilioCapability(HIFIConstants.TWILIO_ACCOUNT_SID, HIFIConstants.TWILIO_AUTH_TOKEN)
capability.allowClientIncoming(HIFIConstants.TWILIO_CLIENT_INCOMING);
 capability.allowClientOutgoing(applicationSid);
token = capability.generateToken();
```

The logs shown when it tries to connect are the following. It seems to me that the logs are API endpoint calls. I am copying the payload as that is where all the information is to determine what the issue is. 

Below are the payloads that are always sent on the same enpoint below. The response to all these calls is always "OK". 

https://eventgw.us1.twilio.com/v4/EndpointEvents

{"publisher":"twilio-js-sdk","group":"connection","name":"outgoing","timestamp":"2022-02-12T05:16:37.183Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","preflight":false},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"network-information","name":"network-change","timestamp":"2022-02-12T05:16:37.188Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","downlink":10,"effective_type":"4g","rtt":50},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"get-user-media","name":"succeeded","timestamp":"2022-02-12T05:16:37.807Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","data":{"audioConstraints":{"audio":true}}},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"signaling-state","name":"have-local-offer","timestamp":"2022-02-12T05:16:37.854Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"new","timestamp":"2022-02-12T05:16:37.859Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-gathering-state","name":"gathering","timestamp":"2022-02-12T05:16:37.865Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.873Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"172.27.32.1","is_remote":false,"port":51986,"priority":2122260223,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.878Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"192.168.1.2","is_remote":false,"network-cost":10,"port":51987,"priority":2122194687,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.967Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"172.27.32.1","is_remote":false,"port":9,"priority":1518280447,"protocol":"tcp","related_address":null,"related_port":null,"tcp_type":"active","transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.971Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"192.168.1.2","is_remote":false,"network-cost":10,"port":9,"priority":1518214911,"protocol":"tcp","related_address":null,"related_port":null,"tcp_type":"active","transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-gathering-state","name":"complete","timestamp":"2022-02-12T05:16:37.975Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"outgoing-ringing","timestamp":"2022-02-12T05:16:38.131Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","hasEarlyMedia":false},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-connection-state","name":"checking","timestamp":"2022-02-12T05:16:38.555Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"settings","name":"edge","timestamp":"2022-02-12T05:16:38.559Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","edge":"singapore"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"signaling-state","name":"stable","timestamp":"2022-02-12T05:16:38.561Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"accepted-by-remote","timestamp":"2022-02-12T05:16:38.568Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"settings","name":"codec","timestamp":"2022-02-12T05:16:38.573Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","codec_params":"","selected_codec":"PCMU/8000"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"pc-connection-state","name":"connecting","timestamp":"2022-02-12T05:16:38.608Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"selected-ice-candidate-pair","timestamp":"2022-02-12T05:16:38.665Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","local_candidate":{"candidate_type":"prflx","deleted":false,"ip":"223.229.188.54","is_remote":false,"network-cost":10,"port":51987,"priority":1853759231,"protocol":"udp","related_address":"192.168.1.2","related_port":51987,"tcp_type":null,"transport_id":""},"remote_candidate":{"candidate_type":"host","deleted":false,"ip":"3.1.77.190","is_remote":true,"port":15920,"priority":2130706431,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":""}},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"connecting","timestamp":"2022-02-12T05:16:38.668Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-connection-state","name":"connected","timestamp":"2022-02-12T05:16:38.671Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"connected","timestamp":"2022-02-12T05:16:38.832Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"pc-connection-state","name":"connected","timestamp":"2022-02-12T05:16:38.835Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"disconnected-by-remote","timestamp":"2022-02-12T05:16:39.552Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"closed","timestamp":"2022-02-12T05:16:39.564Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

What I see in the last two logs is that it is disconnected by remote and then closes. I am not disconnecting the call myself. 

Also the twilio signal URL 

wss://chunderw-vpc-gll.twilio.com/signal

that is called by the library stays on PENDING on the chrome network inspect for a long time and then moves to status UPGRADE. It's status is always 101 (switching?) but it never becomes 200. Don't know if this issue is related to the call not being able to connect.. 

I have been reading up on the scp paragraph in the [github page](https://github.com/twilio/twilio-voice.js)  It talks of configuring SCP so that the signal URL is allowed to go through. Given the logs I have above that is able to register the device, I am wondering if I need to add this scp tag on the page. In fact, I did try it but it gave me other errors like "CSP tag cannot be used in meta please use header" and "Refused to load the script because it violates the following Content Security Policy directive". This is possible as I am not using the write syntax but just wanted to mention what I have tried. 

I don't get any other logs in the browser except the endpoint logs. The logs on twilio account website don't show up for the number I call to which I suppose is because the call has not gone through. The number I call is a US number and I have tried calling the number adding country code 1 and +1 and both fail. They fail with ringing tone and disconnect tone immediately after. 

My previous version works. And for this migration, I have followed this [post](https://www.twilio.com/docs/voice/sdks/javascript/migrating-to-js-voice-sdk-20). 

Hence, If anyone can throw some light on what I am missing (during migration) or doing wrong, would appreciate the help 

 - [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

If possible, please include debug logs. You can set the log level for
twilio-video.js by calling `connect` with the `logLevel` property, e.g.

    connect(token, { logLevel: 'debug' })

Then you can copy and paste or save the logs from your browser's console.

-->

TODO

**Software versions:**

 - [x] Browser(s): chrome 97
 - [x] Operating System: Windows 10
 - [x] twilio-video.js: 2.1 using it as a single JS file from /dist folder as mentioned in the migration documentation. 
 - [x] Third-party libraries:VueJS

**Comment by ricksonmenezes** [comment]
Thanks for moving it Manjesh. Any idea what am I doing wrong? I have seen another issue on twilio-android saying 'disconnected by remote' which also appears to me on web(JS). In android app forum, they keep talking about their Twiml app not giving the proper response. In my case, our account does not have a TwiML app, it still works on production(perhaps  it uses SID, authtoken, secret etc and that's good enough?). 

So while I have migrated the JS library to 2.1, need to know if there is something I am missing (say like TwiMl app is mandatory if you upgrade Js from twilio-client to twilio-voice SDK but it wasn't mentioned in the docs?) due to which I get 'disconnected-by-remote'.

**Comment by mhuynh5757** [comment]
Hi @ricksonmenezes , is it possible for you to migrate to Access Tokens instead of Capability Tokens? The latter are deprecated and no longer officially supported. This may solve your issue. There is also a snippet in the README for this repo that goes over some settings if you are having issues with Content Security Policy.

**Comment by gwing33** [comment]
@manjeshbhargav after seeing your comment here, I decided to upgrade our token. When I switch out the capability token for the access token, calling no longer works. Is there an upgrade guide for what needs to be done here? I wasn't able to find one on Twilio docs.

**Comment by ricksonmenezes** [comment]
> Hi @ricksonmenezes , is it possible for you to migrate to Access Tokens instead of Capability Tokens? The latter are deprecated and no longer officially supported. This may solve your issue. There is also a snippet in the README for this repo that goes over some settings if you are having issues with Content Security Policy.

@mhuynh5757 Thank you for your response. We eventually managed to resolve it by using a sample java app voice-javascript-sdk-quickstart-java that had twilio.sdk twilio 8.1.1. that had a twilio quickstart JS file that calls twilio voice sdk 2.1.1 which is in a singleJS file.

**Comment by mhuynh5757** [comment]
Thanks, we'll file a ticket to see if we can improve our docs, will close this ticket now.

**Comment by Ken-vdE** [comment]
I'm having the same issue. However, adding device.register() did not help, unfortunately...
This is the entire log, creating a device and calling connect() to create an outgoing call (no errors in Twilio console, just immediate disconnect and no call is created):
![image](https://user-images.githubusercontent.com/15888558/182306381-9b3e91ff-5791-4f12-979b-f82f0fee9573.png)

Can anyone help me?

**Comment by bld010** [comment]
@Ken-vdE Hi there! Developer Educator here ... Let's try to figure this out. 

First thing I would check is the AccessToken. Decode it at [jwt.io](https://jwt.io) and make sure it matches what is listed on the [AccessToken docs page](https://www.twilio.com/docs/iam/access-tokens#create-an-access-token-for-voice). 

Next, I would check in the Twilio Console's Call Logs and see if a call is being created. (This is different from the Twilio.Call instance that is returned within your client-side application) Do you see a new call? Or do you see a parent call and a child call? Or no calls at all? 

If there are no calls in the call log, that tells me something is probably wrong with the AccessToken. 

If you see one call (a parent call), but no child call, that tells me something may be wrong with your TwiML. In these cases, your application is able to connect to Twilio (so Twilio creates a new Call Resource that you can see in your Call Logs), but Twilio may not be receiving valid TwiML instructions in response to the HTTP request that it sends to your TwiML App's Voice Configuration Request URL. Check that your TwiML App's Voice URL is configured properly, and that the TwiML you expect is being returned from that endpoint (this is probably <Dial><Number> or <Dial><Client>). If you're passing a ConnectOptions argument in device.connect(), make sure that the parameter names you're sending over from your frontend match the parameter names you're expecting in your TwiML endpoint. 

If this doesn't help, can you screenshot any errors you're getting in your Twilio Console? If you see a parent call in your Console's call logs, expand all of the information in there and send a screenshot.

**Comment by Ken-vdE** [comment]
@bld010 Hi Brianna,
Thank you for your quick response. Unfortunately I've already tried your troubleshooting steps.

I've checked the used token. When I decode it, it looks exactly as it should. The access keys and grants are correct. Is there anything specific that you think I should triple check?

No call logs have been created, which seems logical because it does not look like my application actually creates a call because it is 'disconnected' immediately.

Also no error logs are added with regards to incorrect parameters being sent to my app through a webhook, because no webhook is fired (I don't think anything is sent to Twilio when I try to start the call).

The @twilio/voice-sdk plugin does make 3 POST requests to `https://eventgw.us1.twilio.com/v4/EndpointEvents` with these payloads:
![image](https://user-images.githubusercontent.com/15888558/182546287-33c04e92-711e-4942-9c23-8fec7e0c1c0c.png)
![image](https://user-images.githubusercontent.com/15888558/182546344-a9cc5d40-0ea7-4ca2-9039-df2727a61547.png)
![image](https://user-images.githubusercontent.com/15888558/182546376-ba925ec8-f789-4557-a284-5ba4753e7a34.png)

Does that help?

**Comment by bld010** [comment]
@Ken-vdE Hmm... Are you able to show me how you're instantiating your device and how you're calling device.connect()?

Also, are you using Vue? Are you saving the device in the Vue component's data object? Are you passing the device as a prop? 

I notice in your screenshot that there are things like "Device3" and "Call2" ... are you naming those or is that happening automatically?

**Comment by Ken-vdE** [comment]
@bld010 You pointed me in the right direction!
Yes, I was using Vue. I created a device and passed it to a Vuex store module. Then I retrieved it using a Vuex getter in another Vue component. Using the 'gotten' device, I tried to call `.connect()`.

I refactored the code to create and return the device within a single file and removed any Vue bindings so Vue proxies etc. aren't able to mess with the device object. The only way the application can create and retrieve the device is through 'native' javascript:
```javascript
import { Device } from '@twilio/voice-sdk';

/**
 * Do NOT use this object in Vue. Vue wraps proxies around it and that causes issues.
 *
 * @type { import('@twilio/voice-sdk').Device }
 */
let device = null;

/**
 * @param {String} token
 * @returns {Device}
 */
export function createDevice(token) {
    return device = new Device(token, {
        // See https://www.twilio.com/docs/voice/sdks/javascript/twiliodevice#deviceoptions
        logLevel: 0,
        // Set Opus as our preferred codec. Opus generally performs better, requiring less bandwidth and
        // providing better audio quality in restrained network conditions.
        codecPreferences: ['opus', 'pcmu'],
    });
}

/**
 * @returns { import('@twilio/voice-sdk').Device }
 */
export function getDevice() {
    return device;
}
```


Thank your for your help!

**Comment by bld010** [comment]
@Ken-vdE Happy to help!

**Comment by MichaelSemchenko** [comment]
got same issue with the React. Implemented exactly the same logic with quickstart (what works fine with my token).
But got in console in my app:
```
The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. 

Warning: This browser does not support audio output selection.
```

Perhaps someone know how to solve this?
